src/main/java/liv/processor/Parser.java

```diff
@@ -0,0 +1,197 @@
+package liv.processor;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import liv.exception.LivException;
+import liv.task.Deadline;
+import liv.task.Event;
+import liv.container.TaskList;
+import liv.task.TodoTask;
+
+/**
+ * Process the user command input.
+ */
+public class Parser {
+    /**
+     * The input format for date time.
+     * Uses for deadline and event task command.
+     */
+    private static final DateTimeFormatter INPUT_PATTERN = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    /**
+     * Creates a command object from the user input.
+     * @param input The input of the user.
+     * @return A command object with the type depends on the users input.
+     * @throws LivException If the command cannot is not valid.
+     */
+    public static Command parse(String input) throws LivException {
+        if (input.equalsIgnoreCase("bye")) {
+            return parseByeCommand();
+        } else if (input.equalsIgnoreCase("list")) {
+            return parseListCommand();
+        } else if (input.startsWith("mark")) {
+            return parseMarkOrUnmarkCommand(input, true);
+        } else if (input.startsWith("unmark")) {
+            return parseMarkOrUnmarkCommand(input, false);
+        } else if (input.startsWith("delete")) {
+            return parseDeleteCommand(input);
+        } else if (input.startsWith("find")) {
+            return parseFindCommand(input);
+        } else if (input.startsWith("todo")) {
+            return parseTodoCommand(input);
+        } else if (input.startsWith("deadline")) {
+            return parseDeadlineCommand(input);
+        } else if (input.startsWith("event")) {
+            return parseEventCommand(input);
```

Comment:
I like how the clear naming of functions in the Parser class makes it easy to read when being called.